package com.day4.ex_handling;

import java.io.*;
import java.util.InputMismatchException;
import java.util.Scanner;

public class A_NeedOfExHandling {
	public static void main(String[] args) {
		
		System.out.println("--------------------");

		Scanner scanner=null;
		try{
			//risky code
			int i, j, k;
			 scanner = new Scanner(System.in);
			System.out.println("PE value of first no");
			i = scanner.nextInt();
			j = scanner.nextInt();
			k = i / j;
			System.out.println("k : " + k);
			
		}catch(ArithmeticException e){
			System.out.println("divide by zero is not possible");
			
		}catch(InputMismatchException e){
			System.out.println("u should only enter ints");
			
		}finally{
			//always going to execute wheter ex comes on not?
			scanner.close();
		}
		
		System.out.println("-------------------");
		
		
		//Attempt 2 : better handling the ex... but about the closing the resouces?
//		System.out.println("--------------------");
//
//		Scanner scanner=null;
//		try{
//			//risky code
//			int i, j, k;
//			 scanner = new Scanner(System.in);
//			System.out.println("PE value of first no");
//			i = scanner.nextInt();
//			j = scanner.nextInt();
//			k = i / j;
//			System.out.println("k : " + k);
//			scanner.close();
//		}catch(ArithmeticException e){
//			System.out.println("divide by zero is not possible");
//			scanner.close();
//		}catch(InputMismatchException e){
//			System.out.println("u should only enter ints");
//			scanner.close();
//		}
//		
//		System.out.println("-------------------");
		
		
		
		
		
		
		
		
		
//	     System.out.println("--------------------");
//
//			int i, j, k;
//			Scanner scanner = new Scanner(System.in);
//			System.out.println("PE value of first no");
//			i = scanner.nextInt();
//			j = scanner.nextInt();
//			k = i / j;
//			System.out.println("k : " + k);
//			
//			System.out.println("-------------------");
			
	}

}
